test_config/test_config.go

```diff
@@ -73,6 +79,41 @@ func setABIPath() {
 	ABIFilePath = gp + "/src/github.com/makerdao/vulcanizedb/pkg/eth/testing/"
 }
 
+func NewTestBlockchain() (core.BlockChain, error) {
+	ipc := TestClient.IPCPath
+	// If we don't have an ipc path in the config file, check the env variable
+	if ipc == "" {
+		configErr := viper.BindEnv("url", "CLIENT_IPCPATH")
+		if configErr != nil {
+			return nil, fmt.Errorf("unable to bind url to CLIENT_IPCPATH env var %w", configErr)
+		}
+		ipc = viper.GetString("url")
+	}
+
+	rpcClient, ethClient, clientErr := getClients(ipc)
+	if clientErr != nil {
+		return nil, fmt.Errorf("failed to get test clients: %w", clientErr)
+	}
+
+	return getBlockChain(rpcClient, ethClient)
+}
+
+func getClients(ipc string) (client.RpcClient, *ethclient.Client, error) {
+	raw, err := rpc.Dial(ipc)
+	if err != nil {
+		return client.RpcClient{}, &ethclient.Client{}, err
+	}
+	return client.NewRpcClient(raw, ipc), ethclient.NewClient(raw), nil
+}
+
+func getBlockChain(rpcClient client.RpcClient, ethClient *ethclient.Client) (core.BlockChain, error) {
```

Comment:
Why return an error here if this function can never error?

Reply:
nice catch! i'll update that